src/preact/base-element.js

```diff
@@ -725,6 +728,26 @@ function collectProps(Ctor, element, ref, defaultProps) {
   return props;
 }
 
+/**
+ * Clones an Element into a VNode.
+ * @param {!Element} element
+ * @return {!PreactDef.Renderable}
+ */
+function createVNode(element) {
+  const {attributes} = element;
+  const props = {key: element};
+  for (let i = 0; i < attributes.length; i++) {
+    const {name, value} = attributes[i];
+    props[name] = value;
+  }
+  const children = [];
+  for (let i = 0; i < element.children.length; i++) {
+    // TODO(alanorozco): Could these be text nodes?
```

Comment:
But, yes, you could get text nodes as well. But you'd have to use `element.childNodes` instead of `element.children`.